client/hwlib/src/collectors/hardware_info.rs

```diff
@@ -78,10 +75,12 @@ impl TryFrom<(&SMBiosProcessorInformation<'_>, &Path)> for Processor {
 
     fn try_from(value: (&SMBiosProcessorInformation, &Path)) -> Result<Self> {
         let (processor_info, max_cpu_frequency_filepath) = value;
-        let cpu_id = CpuId::from_smbios(processor_info)?;
+        let cpu_id = processor_info
+            .processor_id()
+            .ok_or_else(|| anyhow!("processor ID not found"))?;
```

Comment:
This may not be the case, but I think there's a slightly strange interaction here. I'm assuming that having `[0; 8]` means 'unknown', but here we are getting an error that the processor ID hasn't been found (somehow these seem similar to me). I think 'not found' may be slightly ambiguous here: is this not found as in not found locally, or not found as in found locally but not recognised?

I think the issue may be that `.processor_id` returns an `Option` and the `identifier` field has behaviour similar to an `Option`, but if these were given the same type, it would be easy for someone to accidentally come along and put `identifier: processor_info.processor_id()` and miss out the important case of the processor ID missing. I think this could be fixed by adding a new enum to represent whether the ID is known/unknown, which specifically _doesn't_ implement `From<Option<_>>`. (This would then force a manual conversion and hence for the locally-missing case to be explicitly considered)

This may be overkill though, but I just have a slightly odd feeling

Reply:
Updated in bb02da2